boilerplate/openshift/golang-osd-operator/csv-generate/common-generate-operator-bundle.py

```diff
@@ -69,43 +203,81 @@
             "version": crd["spec"]["version"]
         }
     )
+# These will be written to the bundle at the end along with generic resources
 
-csv['spec']['install']['spec']['clusterPermissions'] = []
+## Process [Cluster]Role[Binding]s (TODO: Match up ServiceAccounts)
+# Role and ClusterRole are processed similarly
+rolekind_csvkey_bindingmap = (
+    ('ClusterRole', 'clusterPermissions', crb_by_cr),
+    # NOTE: The schema supports `permissions` for Roles, but OLM may not. For
```

Comment:
> but OLM may not

Has this been confirmed?

Reply:
No. The point of this comment is that we need to find out. For now we're doing things the same way as the operators have been. IMO we shouldn't try to change that in this PR.

Reply:
Upon further thought & discussion:
Whether or not OLM actually supports the `permissions` allowed by the schema, it doesn't have a way to manage cross-namespace bindings -- and that's why we're seeing these files being laid down explicitly in the bundle.

We should possibly queue up a TODO to introspect RoleBindings we come across, determine whether they reference owned ServiceAccounts, and include them in `permissions` if they do. But for now, unless we come across a case of this in an existing operator, let's leave it as written (*all* Role[Binding]s go into the bundle directly).

Reply:
I confirmed that none of our CSV templates reference `permissions` 

```
find ./* -type f -regextype egrep -regex ".*csv-template.*" -exec grep -iH permissions: {} \;
./aws-account-operator/config/templates/aws-account-operator-csv-template.yaml:      clusterPermissions:
./aws-efs-operator/config/templates/aws-efs-operator-csv-template.yaml:      clusterPermissions:
./certman-operator/config/templates/certman-operator-csv-template.yaml:      clusterPermissions:
./custom-domains-operator/config/templates/custom-domains-operator-csv-template.yaml:      clusterPermissions:
./deadmanssnitch-operator/config/templates/deadmanssnitch-operator-csv-template.yaml:      clusterPermissions:
./gcp-project-operator/config/templates/gcp-project-operator-csv-template.yaml:      clusterPermissions:
./managed-upgrade-operator/config/templates/managed-upgrade-operator-csv-template.yaml:      clusterPermissions:
./must-gather-operator/config/templates/must-gather-operator-csv-template.yaml:      clusterPermissions:
./osd-metrics-exporter/config/templates/osd-metrics-exporter-csv-template.yaml:      clusterPermissions:
./rbac-permissions-operator/config/templates/rbac-permissions-operator-csv-template.yaml:      clusterPermissions:
./splunk-forwarder-operator/config/templates/splunk-forwarder-operator-csv-template.yaml:      clusterPermissions:
```

Managed velero seems to have two CSV templates 

```
./managed-velero-operator/config/templates/managed-velero-operator-csv-template.yaml:      clusterPermissions:
./managed-velero-operator/config/templates/managed-velero-operator-csv-template.yaml:      permissions:
```

And configure-alertmanager-operator needs the path of the template changed and the permissions are added in the generate script [here](https://github.com/openshift/configure-alertmanager-operator/blob/master/build/generate-operator-bundle.py#L95)

```
cat build/templates/csv.yaml.tmpl
apiVersion: operators.coreos.com/v1alpha1
kind: ClusterServiceVersion
```